src/mlpack/core/data/load_arff_impl.hpp

```diff
@@ -265,16 +298,30 @@ void LoadARFF(const std::string& filename,
           {
             // Okay, it's not NaN or inf.  If it's '?', we issue a specific
             // error, otherwise we issue a general error.
-            std::stringstream error;
             std::string tokenStr = token.str();
             Trim(tokenStr);
             if (tokenStr == "?")
-              error << "Missing values ('?') not supported, ";
+            {
+              if (fatal)
+                Log::Fatal << "Missing values ('?') not supported, "
+                    << std::endl;
+              else
+                Log::Warn << "Missing values ('?') not supported, "
+                    << std::endl;
+            }
             else
-              error << "Parse error ";
-            error << "at line " << (headerLines + row) << " token " << col
-                << ": \"" << tokenStr << "\".";
-            throw std::runtime_error(error.str());
+            {
+              std::stringstream error;
+              error << "Parse error " << "at line " <<
+                  (headerLines + row) << " token " << col << ": \""
+                  << tokenStr << "\"." << std::endl;
```

Comment:
```suggestion
                  << tokenStr << "\".";
```
No need for this newline, since we send another one down below.